src/main/utils/shell-env.ts

```diff
@@ -31,14 +31,20 @@ let shellSyncFailed = false;
 export async function shellEnv(shell?: string, timeout?: number) : Promise<EnvironmentVariables> {
   let envVars = {};
 
+  timeout ??= 0;
+
   if (!shellSyncFailed) {
     try {
-      envVars = await Promise.race([
-        shellEnvironment(shell),
-        new Promise((_resolve, reject) => setTimeout(() => {
-          reject(new Error("Resolving shell environment is taking very long. Please review your shell configuration."));
-        }, timeout))
-      ]);
+      if (timeout > 0 ) {
+        envVars = await Promise.race([
+          shellEnvironment(shell),
+          new Promise((_resolve, reject) => setTimeout(() => {
+            reject(new Error("Resolving shell environment is taking very long. Please review your shell configuration."));
+          }, timeout))
+        ]);
+      } else {
+        envVars = await shellEnvironment(shell);
```

Comment:
Why would we want to not timeout?

Reply:
To provide generic behaviour? (it is how we were using it before--the `shell-env` package does not support a timeout). Otherwise we need to force a minimum timeout. Or we could make `shellEnv()` less generic, just hard-code it with a 5s timeout.

Reply:
Why do we need to force a minimum timeout? The above code handles the 0s timeout excellently (it will always fail).

Reply:
How is that useful?

Reply:
Safer I would argue. Since you would have to explicitly pass in `Number.PositiveInfinity`

Reply:
I think we want always timeout... but if first call does timeout we should have a way to try again.

Reply:
I thought the user would have to restart lens to try the shell sync again. This is what I do with vs code, though maybe I missed a way to do it without restarting.

Reply:
> Safer I would argue. Since you would have to explicitly pass in `Number.PositiveInfinity`
But it is unnecessary to create a timer in this case. The timer would never get cleared, if the `shellEnv` function blocked

Reply:
nvm, we always want to timeout